op-supervisor/supervisor/backend/db/db.go

```diff
@@ -481,6 +481,12 @@ func (db *DB) readCanonicalHash(entryIdx int64) (canonicalHash, error) {
 	return newCanonicalHashFromEntry(data)
 }
 
+func (db *DB) GetLogBlockNum() uint64 {
+    db.rwLock.RLock()
+    defer db.rwLock.RUnlock()
+    return db.lastEntryContext.blockNum
+}
```

Comment:
It won't quite be this simple unfortunately because we can't know if all the logs from the last block were actually written successfully before shutdown.  So to ensure we always recover safely, we'll need to get the last checkpoint that was written (could use `ClosestBlockInfo` and pass `math.maxUInt64`) then `Rewind` the db to the block prior to that block and start from there. That guarantees we will always roll back at least one block so we know we're always starting from a fully written block.

Once we have that basic version a follow up PR can add in logic to check that the returned block info is still for a canonical block and if not, try again with the previous checkpoint block until it either finds a canonical block or reaches genesis (ie has no more checkpoint blocks). I'd definitely do that as a second PR though and that should then complete https://github.com/ethereum-optimism/optimism/issues/11027

Reply:
Given this only happens once at startup I'd just keep things simple and use `maxUInt64` for now.  We'll need to do a round of optimising the database lookups by adding some caching which may help and in particular in this case there's a really simple optimisation to detect when the requested block number is after the last entry and then just go straight to the last checkpoint entry.  The focus for now though is just to get to a minimal useful implementation as fast as possible so we can actually start integrating it with op-node and verifying it all hangs together correctly. After that we'll then have some more real world data about the types of queries it needs to handle and can optimise based on that.